packages/block-library/src/blocks/header/style.scss

```diff
@@ -1,61 +1,156 @@
 @import 'scss/menu-toggle';
 
 @media only screen and (min-width: 1000px) {
-	// Nova Blocks hacks
-	.site-header {
-		// styles from Nova Blocks should only affect blocks created by it
-		@import 'scss/menu';
-	}
-
-	.site-header__content {
-		display: grid;
-		align-items: center;
-		grid-gap: 1em;
-
-		> * {
-			justify-content: center;
-		}
-
-		> :first-child {
-			justify-content: flex-start;
-		}
-
-		> :last-child {
-			justify-content: flex-end;
-		}
-
-		> .wp-block-novablocks-navigation {
-			> .menu,
-			> .menu > ul {
-				justify-content: inherit;
-			}
-		}
+  // Nova Blocks hacks
+  .site-header--default {
+    // styles from Nova Blocks should only affect blocks created by it
+    @import 'scss/menu';
+  }
+  
+  .site-header__content {
+    
+    > .wp-block-novablocks-navigation {
+      > .menu,
+      > .menu > ul {
+        justify-content: inherit;
+      }
+    }
+    
+    &.alignfull[class] {
+      margin-left: 0;
+      margin-right: 0;
+    }
+  }
+  
+  .c-menu-toggle__checkbox,
+  .c-menu-toggle {
+    display: none;
+  }
+  
+  // Fix Header layout until
+  // user hit save after migration.
+ .site-header--is-old {
+   
+   &.site-header--logo-left {
+     
+     .site-header__content {
+       grid-template-columns: auto 1fr;
+     }
+   }
+    .site-header__content {
+      display: grid;
+      grid-template-columns: 1fr auto 1fr;
+      
+      > :last-child {
+        justify-content: flex-end;
+      }
+    }
+  }
+}
 
-		.site-header--logo-left & {
-			grid-template-columns: auto 1fr;
-		}
+@media not screen and (min-width: 1000px) {
+  @import 'scss/mobile';
+  @import 'scss/mobile-transitions';
+}
 
-		.site-header--logo-center & {
-			grid-template-columns: 1fr auto 1fr;
-		}
+.site-header-sticky {
+  position: fixed;
+  top: var(--admin-bar-height, 0);
+  
+  width: 100%;
+  z-index: 1100;
+  
+  opacity: 0;
+  pointer-events: none;
+  
+  transition: opacity .15s ease-in;
+  
+  &--is-visible {
+    opacity: 1;
+    pointer-events: auto;
+  }
+}
 
-		.site-header--logo-right & {
-			grid-template-columns: 1fr auto;
-		}
+.site-header {
+  
+  &--is-sticky {
```

Comment:
`.site-header--is-sticky` would be more clearer since `.site-header` isn't used at all. Also, `--sticky` would be a better modifier for the `site-header` class, while `is-sticky` would make more sense as a separate class.

`.site-header-sticky` and `.site-header--is-sticky` create some confusion at this point while they even have the same rules:
```css
    position: fixed;
    top: var(--admin-bar-height, 0);
```

I would look into renaming / reusing more classes to have classes with more precise scopes. I would keep `site-header` on all site-headers and maybe change the JS selector to something like `js-site-header`, and target the main site header with something like `site-header--main` or `site-header--primary` or `site-header--static`.

I would use the `site-header--sticky` class for making any of the two sticky, and then hide on mobile only the header with the class `site-header--secondary` or `site-header--fixed`. The classes I used below are only for demonstration purposes. The `site-header--default` class may be useless in the example below but I wanted just make a point about consistency in naming.

```html
<!-- main header -->
<div class="site-header site-header--default site-header--static [site-header--sticky] js-site-header">

<!-- sticky header -->
<div class="site-header site-header--default site-header--fixed site-header--sticky">